featuretools/primitives/standard/transform/time_series/rolling_outlier_count.py

```diff
@@ -0,0 +1,120 @@
+import numpy as np
+import pandas as pd
+from woodwork import init_series
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import Datetime, Double
+
+from featuretools.primitives.base.transform_primitive_base import TransformPrimitive
+from featuretools.primitives.standard.transform.time_series.utils import (
+    apply_rolling_agg_to_series,
+)
+
+
+class RollingOutlierCount(TransformPrimitive):
+    """Determines how many values are outliers over a given window.
+
+    Description:
+        Given a list of numbers and a corresponding list of
+        datetimes, return a rolling count of outliers within the numeric values,
+        starting at the row `gap` rows away from the current row and looking backward
+        over the specified window (by `window_length` and `gap`). Values are deemed
+        outliers using the IQR method, computed over the whole series.
+        Input datetimes should be monotonic.
+
+    Args:
+        window_length (int, string, optional): Specifies the amount of data included in each window.
+            If an integer is provided, it will correspond to a number of rows. For data with a uniform sampling
+            frequency, for example of one day, the window_length will correspond to a period of time, in this case,
+            7 days for a window_length of 7.
+            If a string is provided, it must be one of Pandas' offset alias strings ('1D', '1H', etc),
+            and it will indicate a length of time that each window should span.
+            The list of available offset aliases can be found at
+            https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases.
+            Defaults to 3.
+        gap (int, string, optional): Specifies a gap backwards from each instance before the
+            window of usable data begins. If an integer is provided, it will correspond to a number of rows.
+            If a string is provided, it must be one of Pandas' offset alias strings ('1D', '1H', etc),
+            and it will indicate a length of time between a target instance and the beginning of its window.
+            Defaults to 0, which will include the target instance in the window.
```

Comment:
We should update the default gap to be `1` to be consistent with the changes in this PR: https://github.com/alteryx/featuretools/pull/2282